File: backend/src/app/schemas/campaign.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

# Základní vlastnosti
class CampaignBase(BaseModel):
    name: str
    description: Optional[str] = None

# Schéma pro vytváření kampaně
class CampaignCreate(CampaignBase):
    world_id: int # Při vytváření kampaně musíme specifikovat svět

# Schéma pro úpravu kampaně
class CampaignUpdate(CampaignBase):
    name: Optional[str] = None
    world_id: Optional[int] = None # Možná chceme umožnit přesun kampaně? Zatím ne.

# Schéma pro čtení dat kampaně
class Campaign(CampaignBase):
    id: int
    world_id: int
    owner_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        # orm_mode = True
        from_attributes = True 

File: frontend/src/types/campaign.ts
// src/types/campaign.ts

// Základní struktura dat kampaně vracená z API
export interface Campaign {
  id: number;
  name: string;
  description: string | null;
  world_id: number;
  created_at: string; // Datum jako string (ISO formát)
  updated_at: string; // Datum jako string (ISO formát)
  // Zde můžeme přidat další pole, pokud API vrací např. info o světě
  // world?: { id: number; name: string };
}

// Data potřebná pro vytvoření kampaně
export interface CampaignCreate {
  name: string;
  world_id: number; // Povinné při vytváření
  description?: string | null;
}

// Data povolená pro úpravu kampaně
export interface CampaignUpdate {
  name?: string;
  description?: string | null;
  // world_id zde neumožňujeme měnit přes standardní update
} 